Engine/DataFeeds/Synchronizer.cs

```diff
@@ -111,12 +113,21 @@ public virtual IEnumerable<TimeSlice> StreamData(CancellationToken cancellationT
                 if (timeSlice.Time != previousEmitTime)
                 {
                     previousEmitTime = timeSlice.Time;
+                    // if we emitted refresh retry
+                    retried = false;
                     yield return timeSlice;
                 }
                 else if (timeSlice.SecurityChanges == SecurityChanges.None)
                 {
-                    // there's no more data to pull off, we're done (frontier is max value and no security changes)
-                    break;
+                    // if the slice has data lets retry just once more... this could happen
+                    // with subscriptions added after initialize using algorithm.AddSecurity() API,
+                    // where the subscription start time is the current time loop (but should just happen once)
```

Comment:
also, I believe this behavior is required via universe selection and noticeable when doing daily selection for daily data to ensure we start processing that new daily data immediately